app/services/link_creator.rb

```diff
@@ -0,0 +1,7 @@
+class LinkCreator
+  class << self
+    def create(entity)
+      LinkedToken.create!(entity: entity, token: SecureRandom.uuid)
```

Comment:
Personally, I might just add a 

```ruby
before_validation :set_token, on: :create

def set_token
  self.token ||= SecureRandom.uuid
end
```

to the model, but I'm also fine with a service.

Reply:
Sweet, I didn't know this was possible. I tend to lean towards services just to limit the surface area of code using AR.

Reply:
I updated the model to use the on create hook! Thanks for the tip.